crates/bevy_winit/src/winit_config.rs

```diff
@@ -1,15 +1,92 @@
-#[derive(Debug, Default)]
+use bevy_utils::Duration;
+
+/// A resource for configuring usage of the `rust_winit` library. By default, the app will run as
+/// fast as possible when in use, and will be limited to 10fps when not focused.
+#[derive(Debug)]
 pub struct WinitConfig {
-    /// Configures the winit library to return control to the main thread after
-    /// the [run](bevy_app::App::run) loop is exited. Winit strongly recommends
-    /// avoiding this when possible. Before using this please read and understand
-    /// the [caveats](winit::platform::run_return::EventLoopExtRunReturn::run_return)
-    /// in the winit documentation.
-    ///
-    /// This feature is only available on desktop `target_os` configurations.
-    /// Namely `windows`, `macos`, `linux`, `dragonfly`, `freebsd`, `netbsd`, and
-    /// `openbsd`. If set to true on an unsupported platform
-    /// [run](bevy_app::App::run) will panic.
+    /// Configures the winit library to return control to the main thread after the
+    /// [run](bevy_app::App::run) loop is exited. Winit strongly recommends avoiding this when
+    /// possible. Before using this please read and understand the
+    /// [caveats](winit::platform::run_return::EventLoopExtRunReturn::run_return) in the winit
+    /// documentation.
+    ///
+    /// This feature is only available on desktop `target_os` configurations. Namely `windows`,
+    /// `macos`, `linux`, `dragonfly`, `freebsd`, `netbsd`, and `openbsd`. If set to true on an
+    /// unsupported platform [run](bevy_app::App::run) will panic.
     pub return_from_run: bool,
+    /// Configures how the winit event loop updates while the window is focused.
+    pub focused_mode: UpdateMode,
+    /// Configures how the winit event loop updates while the window is *not* focused.
+    pub unfocused_mode: UpdateMode,
+}
+impl WinitConfig {
+    /// Configure winit with common settings for a game.
+    pub fn game() -> Self {
+        WinitConfig::default()
+    }
+
+    /// Configure winit with common settings for a desktop application.
+    pub fn desktop_app() -> Self {
+        WinitConfig {
+            focused_mode: UpdateMode::Reactive {
+                max_wait: Duration::from_secs(5),
+            },
+            unfocused_mode: UpdateMode::ReactiveLowPower {
+                max_wait: Duration::from_secs(60),
+            },
+            ..Default::default()
+        }
+    }
+
+    /// Gets the configured `UpdateMode` depending on whether the window is focused or not
+    pub fn update_mode(&self, focused: bool) -> &UpdateMode {
+        match focused {
+            true => &self.focused_mode,
+            false => &self.unfocused_mode,
+        }
+    }
+}
+impl Default for WinitConfig {
+    fn default() -> Self {
+        WinitConfig {
+            return_from_run: false,
+            focused_mode: UpdateMode::Continuous,
+            unfocused_mode: UpdateMode::ReactiveLowPower {
```

Comment:
I don't think falling back to 10 FPS when unfocused is a good default for games:
* Games build logic based on fixed timesteps. For things like physics, ticking at 10 FPS increases the likelihood of "tunneling" and things of that nature. Obviously games should try to protect against tick-time-based discrepancies in behavior because perf issues do happen, but completely insulating against sustained low-tick-rates will be challenging for some types of games.  
* For online games that send and receive network events from inside a game loop, these should still be processed at "reasonable" intervals to encourage proper state syncs. What constitutes "reasonable" is game-specific. I think the only safe assumption to make by default is "developer configured fps limits for active gameplay".
* A common case for "unfocusing games" will be "playing a windowed game and alt-tabbing to another window". In this case, an unfocused game might still be visible and the "perceived quality" of the game will go down if it starts "lagging" when alt-tabbed.

I think throttling _rendering_ (not ticks) by default when unfocused is a slightly more compelling argument. But until we decouple the two (ex: the "actual pipelining" impl would do this), I think we should default to `unfocused_mode = UpdateMode::Continuous` for games.